### DISCUSSION TITLE
How to get a Path's updated coordinates?

### DISCUSSION THREAD

**Discussion by abdussami-tayyab** [fabricjs/fabric.js#8821] [Q&A]
I have created a `Path`, and 3 `Circle` controls to control the left, right and curve value of the `Path` object. So far, it's working correctly:
```
// pseudocode
let path = new fabric.Path('M 100 100 Q 200, 200, 300, 100', { fill: '', stroke: 'black', objectCaching: false });
let lCircle = new fabric.Circle (set to the left side of Path)
let mCircle = new fabric.Circle(set a bit below the Path's middle to control curve)
let rCircle = new fabric.Circle (set to the right side of Path)

// then I have added handlers for the 3 circle controls, and in each handler, get `x` and `y` of the relevant circle, use it to update the `M`
// start/end attribute of the `Path` or the `Q` quadratic curve value. One such handler is below:
rCircle.on("moving", function (e) {
  const x = rCircle.left;
  const y = rCircle.top;

  let oldPath = rCircle.parentPath;
  const segments = oldPath.path;
  const newPath = new fabric.Path("M " + (segments[0][1]) + " " + (segments[0][2]) + " Q " + segments[1][1] + " " + segments[1][2] + " " + (x + 5) + " " + (y + 5), { fill: '', stroke: 'black', objectCaching: false });
  canvas.remove(rCircle.parentPath);
  mCircle.parentPath = lCircle.parentPath = rCircle.parentPath = newPath;
  canvas.add(newPath);
  canvas.renderAll();
});
```
Demo: https://imgur.com/a/xgYPiTV
--
What I want now is to add a handler to the `path`, which updates the values of the circles as well, but the path's segments object is not being updated. And the only thing apart from it I have is the **Bounding Rect** which is also not helpful since it returns the box coords, not the line coordinates.

How can I begin to tackle this problem? I am facing an issue with a simple Line as well which is controlled by circles. I followed the [Quadratic Curve](http://fabricjs.com/quadratic-curve) demo but it modifies the path coords in-place, and the bounding rect does not get updated, as well as it's not movable. Which I want.

**Comment by ShaMan123** [comment]
PerPixelTargetFind is not related.
Search the repo, I remeber seeing a ticket about this with a working demo
